src/components/pages/Signup.js

```diff
@@ -0,0 +1,176 @@
+import React, { useState } from "react";
+import Loader from "react-loader-spinner";
+
+import { useAuthContext } from "../../context/auth-context";
+import { Link, Navigate } from "react-router-dom";
+
+const Signup = () => {
+  const { signupNewUser, isUserLogin } = useAuthContext();
+
+  const [userEmail, setUserEmail] = useState("");
+  const [userPassword, setUserPassword] = useState("");
+  const [userName, setUserName] = useState("");
+  const [serverError, setServerError] = useState("");
+  const [emailError, setEmailError] = useState("");
+  const [passwordError, setPasswordError] = useState("");
+  const [isLoading, setLoading] = useState(false);
+
+  const signupHandler = async (e) => {
+    e.preventDefault();
+    setLoading(true);
+    setServerError("");
+    setEmailError("");
+    setPasswordError("");
+    const res = await signupNewUser(userEmail, userPassword, userName);
+    if (res?.status === 409) {
+      setServerError(
+        res?.data?.message || "Something went wrong, please try again!"
+      );
+      setLoading(false);
+      return;
+    } else {
+      setServerError(
+        res?.data?.errorMessage || "Something went wrong, please try again!"
+      );
+      setLoading(false);
+    }
+  };
+
+  const emailValidator = () => {
```

Comment:
put these functions in separate utility file as these are generally used on both login and sign up components